README.md
# Scala Exercises - Cats library

This repository hosts a content library for the [Scala Exercises](https://www.scala-exercises.org/) platform that includes interactive exercises related to the [Cats library](https://github.com/typelevel/cats) by Typelevel.

## About Scala Exercises

"Scala Exercises" brings exercises for the Stdlib, Cats, Shapeless, and many other great libraries for Scala to your browser. It offers hundreds of solvable exercises organized into several categories covering the basics of the Scala language and its most important libraries.

Scala Exercises is available at [scala-exercises.org](https://scala-exercises.org).

## Contributing

Contributions are welcome! Please join our [Gitter channel](https://gitter.im/scala-exercises/scala-exercises)
to get involved, or visit our [GitHub site](https://github.com/scala-exercises).

## License

Copyright (C) 2015-2020 47 Degrees, LLC.
Reactive, scalable software solutions.
http://47deg.com
hello@47deg.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.


CHANGELOG.md
# Changelog

## [v0.6.8](https://github.com/scala-exercises/exercises-cats/tree/v0.6.8) (2020-08-31)

[Full Changelog](https://github.com/scala-exercises/exercises-cats/compare/v0.6.7...v0.6.8)

📈 **Dependency updates**

- Update definitions, exercise-compiler, ... to 0.6.7 [\#145](https://github.com/scala-exercises/exercises-cats/pull/145) ([47erbot](https://github.com/47erbot))
- Update definitions, exercise-compiler, ... to 0.6.6 [\#143](https://github.com/scala-exercises/exercises-cats/pull/143) ([47erbot](https://github.com/47erbot))
- Update scalacheck-1-14 to 3.2.2.0 [\#142](https://github.com/scala-exercises/exercises-cats/pull/142) ([47erbot](https://github.com/47erbot))
- Update scalatest to 3.2.2 [\#141](https://github.com/scala-exercises/exercises-cats/pull/141) ([47erbot](https://github.com/47erbot))
- Update sbt-mdoc to 2.2.5 [\#140](https://github.com/scala-exercises/exercises-cats/pull/140) ([47erbot](https://github.com/47erbot))

**Closed issues:**

- Eval site is still broken [\#134](https://github.com/scala-exercises/exercises-cats/issues/134)

## [v0.6.7](https://github.com/scala-exercises/exercises-cats/tree/v0.6.7) (2020-08-10)

[Full Changelog](https://github.com/scala-exercises/exercises-cats/compare/v0.6.6...v0.6.7)

📈 **Dependency updates**

- Update definitions, exercise-compiler, ... to 0.6.5 [\#139](https://github.com/scala-exercises/exercises-cats/pull/139) ([47erbot](https://github.com/47erbot))

## [v0.6.6](https://github.com/scala-exercises/exercises-cats/tree/v0.6.6) (2020-08-10)

[Full Changelog](https://github.com/scala-exercises/exercises-cats/compare/v0.6.5...v0.6.6)

📈 **Dependency updates**

- Update definitions, exercise-compiler, ... to 0.6.4 [\#138](https://github.com/scala-exercises/exercises-cats/pull/138) ([47erbot](https://github.com/47erbot))

## [v0.6.5](https://github.com/scala-exercises/exercises-cats/tree/v0.6.5) (2020-08-10)

[Full Changelog](https://github.com/scala-exercises/exercises-cats/compare/v0.6.4...v0.6.5)

📈 **Dependency updates**

- Update sbt-mdoc to 2.2.4 [\#137](https://github.com/scala-exercises/exercises-cats/pull/137) ([47erbot](https://github.com/47erbot))
- Update scalacheck-1-14 to 3.2.1.0 [\#136](https://github.com/scala-exercises/exercises-cats/pull/136) ([47erbot](https://github.com/47erbot))
- Update scalatest to 3.2.1 [\#135](https://github.com/scala-exercises/exercises-cats/pull/135) ([47erbot](https://github.com/47erbot))
- Update sbt-scalafmt to 2.4.2 [\#133](https://github.com/scala-exercises/exercises-cats/pull/133) ([47erbot](https://github.com/47erbot))
- Update scalafmt-core to 2.6.4 [\#132](https://github.com/scala-exercises/exercises-cats/pull/132) ([47erbot](https://github.com/47erbot))
- Update scalafmt-core to 2.6.3 [\#129](https://github.com/scala-exercises/exercises-cats/pull/129) ([scala-steward](https://github.com/scala-steward))
- Update scalafmt-core to 2.6.2 [\#126](https://github.com/scala-exercises/exercises-cats/pull/126) ([scala-steward](https://github.com/scala-steward))

**Merged pull requests:**

- Update EitherSection.scala [\#130](https://github.com/scala-exercises/exercises-cats/pull/130) ([chipmunklimo](https://github.com/chipmunklimo))
- Updated Eval Exercise to take input, spec was off as well [\#127](https://github.com/scala-exercises/exercises-cats/pull/127) ([jeremylaier](https://github.com/jeremylaier))
- Fix minor bug in code snippet example [\#69](https://github.com/scala-exercises/exercises-cats/pull/69) ([tr4rex](https://github.com/tr4rex))

## [v0.6.4](https://github.com/scala-exercises/exercises-cats/tree/v0.6.4) (2020-06-30)

[Full Changelog](https://github.com/scala-exercises/exercises-cats/compare/v0.6.3...v0.6.4)

## [v0.6.3](https://github.com/scala-exercises/exercises-cats/tree/v0.6.3) (2020-06-29)

[Full Changelog](https://github.com/scala-exercises/exercises-cats/compare/v0.6.2...v0.6.3)

📈 **Dependency updates**

- Update sbt to 1.3.13 [\#121](https://github.com/scala-exercises/exercises-cats/pull/121) ([scala-steward](https://github.com/scala-steward))
- Update sbt-mdoc to 2.2.3 [\#120](https://github.com/scala-exercises/exercises-cats/pull/120) ([scala-steward](https://github.com/scala-steward))
- Update scalafmt-core to 2.6.1 [\#119](https://github.com/scala-exercises/exercises-cats/pull/119) ([scala-steward](https://github.com/scala-steward))
- Update scalacheck-1-14 to 3.2.0.0 [\#118](https://github.com/scala-exercises/exercises-cats/pull/118) ([scala-steward](https://github.com/scala-steward))
- Update scalatest to 3.2.0 [\#117](https://github.com/scala-exercises/exercises-cats/pull/117) ([scala-steward](https://github.com/scala-steward))
- Update sbt-mdoc to 2.2.2 [\#111](https://github.com/scala-exercises/exercises-cats/pull/111) ([scala-steward](https://github.com/scala-steward))
- Update scalatest to 3.1.2 [\#103](https://github.com/scala-exercises/exercises-cats/pull/103) ([scala-steward](https://github.com/scala-steward))

**Merged pull requests:**

- Update definitions, exercise-compiler, ... to 0.6.3 [\#122](https://github.com/scala-exercises/exercises-cats/pull/122) ([scala-steward](https://github.com/scala-steward))
- Update scalafmt-core to 2.6.0 [\#115](https://github.com/scala-exercises/exercises-cats/pull/115) ([BenFradet](https://github.com/BenFradet))
- Update scalafmt-core to 2.5.3 [\#113](https://github.com/scala-exercises/exercises-cats/pull/113) ([BenFradet](https://github.com/BenFradet))
- Prepare repository for next  release and SBT build improvements [\#110](https://github.com/scala-exercises/exercises-cats/pull/110) ([juanpedromoreno](https://github.com/juanpedromoreno))
- Update sbt to 1.3.12 [\#109](https://github.com/scala-exercises/exercises-cats/pull/109) ([scala-steward](https://github.com/scala-steward))
- Replace deprecated kind-projector syntax \[use '\*' instead of '?'\] [\#106](https://github.com/scala-exercises/exercises-cats/pull/106) ([Samehadar](https://github.com/Samehadar))

project/ProjectPlugin.scala
import sbt.Keys._
import sbt._
import com.alejandrohdezma.sbt.github.SbtGithubPlugin

object ProjectPlugin extends AutoPlugin {

  override def trigger: PluginTrigger = allRequirements

  override def requires: Plugins = plugins.JvmPlugin && SbtGithubPlugin

  object autoImport {

    lazy val V = new {
      val cats: String                = "2.1.1"
      val shapeless: String           = "2.3.3"
      val scala: String               = "2.13.2"
      val scalaExercises: String      = "0.6.0"
      val scalacheck: String          = "1.14.3"
      val scalacheckShapeless: String = "1.2.5"
      val scalatest: String           = "3.1.2"
      val scalatestplusScheck: String = "3.1.1.1"
    }

    def dep(artifactId: String) = "org.scala-exercises" %% artifactId % V.scalaExercises

    lazy val exercisesSettings = Seq(
      libraryDependencies ++= Seq(
        dep("exercise-compiler"),
        dep("definitions"),
        "org.typelevel"              %% "cats-core"                 % V.cats,
        "com.chuusai"                %% "shapeless"                 % V.shapeless,
        "org.scalatest"              %% "scalatest"                 % V.scalatest,
        "org.scalacheck"             %% "scalacheck"                % V.scalacheck,
        "org.scalatestplus"          %% "scalacheck-1-14"           % V.scalatestplusScheck,
        "com.github.alexarchambault" %% "scalacheck-shapeless_1.14" % V.scalacheckShapeless
      ),
      addCompilerPlugin("org.typelevel" %% "kind-projector" % "0.11.0" cross CrossVersion.full)
    )
  }

  override def projectSettings: Seq[Def.Setting[_]] =
    Seq(
      organization := "org.scala-exercises",
      organizationName := "47 Degrees",
      organizationHomepage := Some(url("https://47deg.com")),
      scalaVersion := autoImport.V.scala,
      resolvers ++= Seq(
        Resolver.sonatypeRepo("snapshots"),
        Resolver.sonatypeRepo("releases")
      )
    )
}
